api/gen_ai/tests/test_gen_ai.py

```diff
@@ -0,0 +1,120 @@
+import hmac
+import json
+from hashlib import sha256
+from unittest.mock import patch
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from shared.django_apps.core.tests.factories import OwnerFactory
+
+from codecov_auth.models import GithubAppInstallation
+
+PAYLOAD_SECRET = b"testixik8qdauiab1yiffydimvi72ekq"
+VIEW_URL = reverse("auth")
+
+
+def sign_payload(payload, secret=PAYLOAD_SECRET):
+    data = json.dumps(payload, separators=(",", ":")).encode("utf-8")
+    signature = "sha256=" + hmac.new(secret, data, digestmod=sha256).hexdigest()
+    return signature, data
```

Comment:
```suggestion
def sign_payload(payload: bytes, secret=PAYLOAD_SECRET):
    signature = "sha256=" + hmac.new(secret, payload, digestmod=sha256).hexdigest()
    return signature, data
```

I would avoid going from bytes -> string -> bytes with `json.dumps()`. Ordering of the keys isn't guaranteed (afaik), this can result in different signatures. The recommendation when working with payload signatures like this is to always calculate them based on the raw bytes.

Reply:
This is just for the tests as I am mimicking a request. The actual code(tested to verify) does not do any conversions like that.

Reply:
If the actual code isn't doing this, I would model the `sign_payload` after it's same process. Otherwise you may end up with some unexpected test failures.